warehouse/ingest-core/src/main/java/datawave/ingest/util/cache/JobCacheFactory.java

```diff
@@ -0,0 +1,195 @@
+package datawave.ingest.util.cache;
+
+import com.google.common.collect.Maps;
+import datawave.common.io.FilesFinder;
+import datawave.ingest.util.cache.converter.CacheLockConverter;
+import datawave.ingest.util.cache.delete.DeleteJobCache;
+import datawave.ingest.util.cache.lease.JobCacheLockFactory;
+import datawave.ingest.util.cache.load.LoadJobCache;
+import datawave.ingest.util.cache.path.FileSystemPath;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.http.util.Args;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/** The JobCacheFactory will expose the basic functionality to interact and use the job cache */
+public class JobCacheFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JobCacheFactory.class);
+    public static final Comparator<Path> PATH_COMPARATOR = Comparator.comparing(x -> x.toUri().getPath());
+    public static final String DW_JOB_CACHE_TIMESTAMP_DIR_PREFIX = "datawave.job.cache.timestamp.dir.prefix";
+    public static final String DW_JOB_CACHE_TIMESTAMP_PATTERN = "datawave.job.cache.timestamp.pattern";
+    public static final String DW_JOB_CACHE_BASE_PATH = "datawave.job.cache.base.path";
+    public static final String DW_JOB_CACHE_LOCK_FACTORY_MODE = "datawave.job.cache.lock.factory.mode";
+    
+    private final FileSystemPath jobCacheBasePath;
+    private final Pattern timestampPattern;
+    private final JobCacheLockFactory lockFactory;
+    private final String timestampDirPrefix;
+    private final Map<String,String> jobIdToCacheId = Maps.newHashMap();
+    
+    public JobCacheFactory(Configuration conf) throws IOException {
+        Path basePath = new Path(Args.notNull(conf.get(DW_JOB_CACHE_BASE_PATH), DW_JOB_CACHE_BASE_PATH));
+        jobCacheBasePath = new FileSystemPath(basePath.getFileSystem(conf), basePath);
+        
+        timestampPattern = Pattern.compile(Args.notNull(conf.get(DW_JOB_CACHE_TIMESTAMP_PATTERN), DW_JOB_CACHE_TIMESTAMP_PATTERN));
+        timestampDirPrefix = Args.notNull(conf.get(DW_JOB_CACHE_TIMESTAMP_DIR_PREFIX), DW_JOB_CACHE_TIMESTAMP_DIR_PREFIX);
+        
+        String lockFactoryMode = Args.notNull(conf.get(DW_JOB_CACHE_LOCK_FACTORY_MODE), DW_JOB_CACHE_LOCK_FACTORY_MODE);
+        lockFactory = new CacheLockConverter().convert(lockFactoryMode);
```

Comment:
I think this is a cool way to initialize this but the fact that we're delegating through a JCommander converter couples this utility to JCommander now. If we ever moved it around, for instance, we'll need to take JCommander with it.

Could you maybe bump that logic into the LockFactory.Mode instead? Like have a `LockFactory.Mode.fromString()` or something like that? Then have the converter and this both use that utility method.

Reply:
Thanks!  Will update.

Reply:
Or maybe just make an alternative `JobCacheLockFactory.getLockFactory(String mode)` method that will throw an Exception if you give it an invalid String.